lib/fluent/log.rb

```diff
@@ -281,6 +284,7 @@ def event(level, args)
       }
 
       map.each_pair {|k,v|
+        next if (k.to_s == "plugin_type" || k.to_s == "plugin_id") and @optional_header
```

Comment:
Umm..., I think that this line based code is still needed because `RootAgent#shutdown` specifies `type` and `plugin_id`.